src/actions/AddLazyJavaClassAction.java

```diff
@@ -22,37 +28,59 @@ public boolean canPerformAction() {
 
 	@Override
 	public void performAction() {
-		String className = JOptionPane
-				.showInputDialog("Please specify the name of the Java Class");
-		try {
-			Class classObj = Class.forName(className);
-			ClassPaintComponent comp = new ClassPaintComponent(classObj,
-					panel.getWidth() / 2, panel.getHeight() / 2);
-			panel.addPaintComponent(comp);
-			// add action to undo redo manager
-			SharedUndoRedoActionManager.getSharedInstance().pushUndoableAction(new UndoRedoableInterface() {
+		ClassSearchFrame searchFrame = new ClassSearchFrame(this);
+		searchFrame.setDelegate(new ClassSearchFrameDelegateInterface() {
+			
+			@Override
+			public void didSelectClass(String classname) {
+				// TODO Auto-generated method stub
```

Comment:
Please study this class. This is the way that the action gets notified when a class is selected.